### PR TITLE
[MM-38868] Fixes do not disturb submenu item not taking full width

### PR DESCRIPTION
**Pull request by AshishDhama**
#### Summary
Do not disturb submenu item not taking full width
(Note: we need to enable TimeDND feature flag)
Screenshot 2021-09-30 at 2 49 38 PM

#### Ticket Link
Fixes [https://mattermost.atlassian.net/browse/MM-38868](https://mattermost.atlassian.net/browse/MM-38868)

#### Related Pull Requests
NONE

#### Screenshots
|  before  |  after  |
|----|----|
| image-20210927-235911 | Screenshot 2021-09-30 at 2 47 44 PM |

#### Release Note
```release-note
* UI: Fixes Do not disturb submenu item not taking full width
```

### CODE REVIEW

File: components/status_dropdown/status_dropdown.scss [hunk 1]
```diff
@@ -40,6 +40,12 @@
     }
 }
 
+#status-menu-dnd-timed_menuitem {
+    .SubMenuItem.MenuItem {
+        width: 100%;
```
Comment (larkox) [inline components/status_dropdown/status_dropdown.scss:45]: Is this how we do it on the rest of the elements? (in this submenu or in the menu before?)

I wonder if we are using in other places flex properties, or something like that.
My point is just to be coherent among the different elements / files.

**Review by larkox — APPROVED** [review]
LGTM, just a comment regarding potential coherence.

File: components/status_dropdown/status_dropdown.scss (hunk 1 above)
Comment (larkox) [reply components/status_dropdown/status_dropdown.scss:45]: After a second look, I think the problem is not so much about taking 100%, but applying two shades at the same time. Maybe the submenu item has already a background and we are applying an extra background we shouldn't?

**Review by Willyfrog — APPROVED** [review]
other than @larkox concerns, LGTM

File: components/status_dropdown/status_dropdown.scss (hunk 1 above)
Comment (larkox) [reply components/status_dropdown/status_dropdown.scss:45]: I can see a difference in the shades between the element selected in the previous menu, and the one selected in this menu. Not sure if desired. Maybe ask someone from UX?

Comment (AshishDhama) [reply components/status_dropdown/status_dropdown.scss:45]: @larkox problem wasn't colouring, when we were clicking on the item nothing was happening. What you are suggesting also make sense.

**Comment by AshishDhama** [comment]
@larkox Please take a look now, have fixed the hover background color issue and instead of width have used flex-grow.
@nevyangelova Could you please go through this as well. If we can remove background from ``SubMenuItemContainer``.

File: components/status_dropdown/status_dropdown.tsx [hunk 2]
```diff
@@ -328,6 +328,7 @@ export default class StatusDropdown extends React.PureComponent <Props, State> {
                     action: index === 4 ? () => setCustomTimedDnd() : () => setDnd(index),
                 } as any;
             }));
+        console.log('dndSubMenuItems', dndSubMenuItems);
```
Comment (larkox) [inline components/status_dropdown/status_dropdown.tsx:331]: Are both changes on this file for debugging?

**Review by larkox — CHANGES_REQUESTED** [review]
LGTM, just remove the (apparently) debugging code.

**Review by Willyfrog — APPROVED** [review]

File: components/status_dropdown/status_dropdown.tsx (hunk 2 above)
Comment (AshishDhama) [reply components/status_dropdown/status_dropdown.tsx]: Thanks, Done

**Review by larkox — APPROVED** [review]

**Review by nevyangelova — APPROVED** [review]

**Pull request merged**